src/plasmapy/particles/particle_collections.py

```diff
@@ -314,41 +314,79 @@ def insert(self, index, particle: ParticleLike) -> None:
             particle = Particle(particle)
         self.data.insert(index, particle)
 
+    @overload
     def is_category(
         self,
         *category_tuple,
         require: str | Iterable[str] | None = None,
         any_of: str | Iterable[str] | None = None,
         exclude: str | Iterable[str] | None = None,
-    ) -> list[bool]:
+        particlewise: Literal[True] = ...,
+    ) -> bool: ...
+
+    @overload
+    def is_category(
+        self,
+        *category_tuple,
+        require: str | Iterable[str] | None = None,
+        any_of: str | Iterable[str] | None = None,
+        exclude: str | Iterable[str] | None = None,
+        particlewise: Literal[False],
+    ) -> list[bool]: ...
```

Comment:
👍🏻 Nice use of `@overload`!

Reply:
So, I did this when I was trying to solve my issues with the static typing.  It didn't solve that issue, but from what I gather reading comments on StackExchange and Github, it seems this is a clear way to specify the output when it changes based on a `bool` input.

Reply:
That is helpful to know!  

If you run into any mypy errors that feel unduly strict, please feel free to bring it up in #2589.  We're still in the process of figuring out how strict we want the mypy settings to be, since there are tradeoffs between the benefits of type hints compared to the effort needed to implement them.  I suspect that we'll probably need to make mypy somewhat less strict.

Reply:
I had to use `# type: ignore[arg-type]` to resolve the issue here:
```
        if (
            not self.allow_custom_particles
            and isinstance(particle, ParticleList)
            and any(particle.is_category("custom", particlewise=True))  # type: ignore[arg-type]
        ):
```
The third condition can only be called if it is indeed a `ParticleList`, but the mypy check insisted that there would be an issue with possibly return something that is not iterable, which means that the call to `any()` would fail.  It might be too strict for this case?  I'll mention it in the other thread.